File: frontend/src/types/trade.ts
/** Position held in a replay session */
export interface Position {
  id: number;
  session_id: number;
  symbol: string;
  quantity: number;
  average_price: number;
  total_cost: number;
  realized_pnl: number;
  unrealized_pnl: number;
  status: 'open' | 'closed';
  opened_at: string;
  closed_at?: string;
}

/** Completed or open trade */
export interface Trade {
  id: number;
  session_id: number;
  symbol: string;
  entry_date: string;
  entry_price: number;
  exit_date?: string;
  exit_price?: number;
  quantity: number;
  gross_pnl?: number;
  net_pnl?: number;
  pnl_percent?: number;
  initial_stop_loss?: number;
  target_price?: number;
  initial_risk?: number;
  r_multiple?: number;
  holding_candles?: number;
  holding_days?: number;
  status?: string;
  result?: string;
  setup_type?: string;
  mistake_tag?: string;
}

export interface Order {
  id: number;
  session_id: number;
  decision_id: number;
  symbol: string;
  side: 'BUY' | 'SELL';
  order_type: 'MARKET_AT_CLOSE' | 'MARKET_NEXT_OPEN' | 'LIMIT' | 'CUSTOM_PRICE';
  requested_price?: number | null;
  quantity: number;
  capital_percent?: number | null;
  status: 'created' | 'pending' | 'executed' | 'cancelled' | 'rejected';
  created_at: string;
}

File: backend/app/models/trade.py
from sqlalchemy import Column, String, Float, Integer, DateTime, ForeignKey
from sqlalchemy.sql import func
from app.db import Base

class Trade(Base):
    __tablename__ = "trades"

    id = Column(Integer, primary_key=True, autoincrement=True)
    session_id = Column(Integer, ForeignKey("replay_sessions.id"), nullable=False)
    symbol = Column(String, nullable=False)
    
    entry_date = Column(DateTime(timezone=True), nullable=False)
    entry_price = Column(Float, nullable=False)
    exit_date = Column(DateTime(timezone=True), nullable=True)
    exit_price = Column(Float, nullable=True)
    quantity = Column(Float, nullable=False)
    
    gross_pnl = Column(Float, nullable=True)
    net_pnl = Column(Float, nullable=True)
    pnl_percent = Column(Float, nullable=True)
    
    initial_stop_loss = Column(Float, nullable=True)
    target_price = Column(Float, nullable=True)
    initial_risk = Column(Float, nullable=True)
    r_multiple = Column(Float, nullable=True)
    
    holding_candles = Column(Integer, nullable=True)
    holding_days = Column(Integer, nullable=True)
    
    status = Column(String, default='open')
    result = Column(String, default='open')
    setup_type = Column(String, nullable=True)
    mistake_tag = Column(String, nullable=True)
    
    created_at = Column(DateTime(timezone=True), default=func.now())
    updated_at = Column(DateTime(timezone=True), default=func.now(), onupdate=func.now())
